addons/website_sale/controllers/main.py

```diff
@@ -909,6 +915,86 @@ def address(self, **kw):
         render_values.update(self._get_country_related_render_values(kw, render_values))
         return request.render("website_sale.address", render_values)
 
+    @http.route(
+        ['/shop/express'], type='json', methods=['POST'],
+        auth="public", website=True, sitemap=False,
+    )
+    def process_express_checkout(self, **post):
+        """ TODO VCR: do a proper docstring
+        TODO VCR: FIX ADDRESS CHANGING
+        """
+        def _create_or_edit_billing_partner(order, billing_address, edit=False):
```

Comment:
Nested function seems wholly unnecessary, can use a utility method (on the controller object or the model, doesn't really matter).